### DISCUSSION TITLE
Best way to customize new variant for Checkbox

### DISCUSSION THREAD

**Discussion by lgaspari** [mantinedev/mantine#5832] [Q&A] [answered]
Hey folks, I'm currently adding a new variant to the `Checkbox` component similar to the `light` variant for `Button` but I couldn't figure out if I'm doing it the right way since I'm overriding the styles with `!important`.

You may find my contrive example below or in [codesandbox](https://codesandbox.io/p/sandbox/mantine-react-template-forked-ds5cht):

```ts
.root {
  &[data-variant='light'] {
    /* due to inline style declarations, we must use !important to override */
    --checkbox-color: var(--mantine-primary-color-light) !important;
    --checkbox-icon-color: var(--mantine-primary-color-filled) !important;
  }
}
```

```ts
import { Checkbox } from '@mantine/core';
import checkboxClassNames from './Checkbox.module.css';

export default createTheme({
  components: {
    Checkbox: Checkbox.extend({
      classNames: checkboxClassNames,
    }),
  },
});
```

Am I doing it the proper way or is it actually hacky? I wanted to avoid using `color` and `iconColor` because I'm using it in a couple of places and I understood that those changes should be in the theme instead of one-off situations.

Thanks in advance!

**Comment by rtivital** [comment, marked as answer]
Components CSS variables are supposed to be overridden with `vars` – https://mantine.dev/styles/styles-api/#components-css-variables

**Reply by lgaspari** [reply]
Cool, I was looking at a different page for creating my own variant (https://mantine.dev/styles/variants-sizes) but since the CSS variables can be overriden using `vars` it looks like the best way to go. Thanks!
